language/stdlib/modules/libra_system.move

```diff
@@ -310,46 +280,32 @@ module LibraSystem {
         !Vector::is_empty(&get_validator_index_(validators_vec_ref, addr))
     }
 
-    // Discovery methods
-
     fun emit_discovery_set_change() acquires DiscoverySet {
         let discovery_set_ref = borrow_global_mut<DiscoverySet>(0xD15C0);
         Event::emit_event<DiscoverySetChangeEvent>(
             &mut discovery_set_ref.change_events,
             DiscoverySetChangeEvent {
                 new_discovery_set: *&discovery_set_ref.discovery_set,
+                new_validator_set: get_validator_set(),
             },
         );
     }
 
-    public fun get_discovery_address(d: &DiscoveryInfo): address {
-        d.addr
-    }
+    // Functions for testing
 
-    public fun get_discovery_config(d: &DiscoveryInfo): &ValidatorConfig::DiscoveryConfig {
-        &d.discovery_config
-    }
-
-    // Get the DiscoveryInfo for the ith validator
-    public fun get_ith_discovery_info(i: u64): DiscoveryInfo acquires DiscoverySet {
-        let discovery_vec_ref = &borrow_global<DiscoverySet>(0xD15C0).discovery_set;
-        *Vector::borrow(discovery_vec_ref, i)
+    // This getter is only used in tests
+    public fun get_validator_config(addr: address): ValidatorConfig::Config {
+        let validator_set = get_validator_set();
+        let validator_index_vec = get_validator_index_(&validator_set.validators, addr);
+        *&(*Vector::borrow(&validator_set.validators, *Vector::borrow(&validator_index_vec, 0))).config
     }
 
-    // Get the index of the discovery info with address `addr` in `discovery_set`.
-    // Aborts if `addr` is not in discovery set.
-    public fun get_discovery_index(discovery_set: &vector<DiscoveryInfo>, addr: address): u64 {
-        let len = Vector::length(discovery_set);
-        let i = 0;
-        loop {
-            if (get_discovery_address(Vector::borrow(discovery_set, i)) == addr) {
-                return i
-            };
-
-            i = i + 1;
-            if (i >= len) { break };
-        };
-
-        abort 99
+    // This getter is only used in tests
```

Comment:
Might be good to include `FOR_TESTING` or similar in the function name just to prevent accidental use... 

We haven't included functions that are intended to be used only for testing elsewhere in the core modules, so might be good to avoid this if we can (though I can't tell how painful that would be here).

Reply:
Agree, they should be removed... These getters were in this module before (lines 53-69), this PR did not introduce them - I just simplified them a bit and moved down here. Right now the module is tested with move functional tests and these tests can not access fields of the resources - that's why these getters are here. We will need to write e2e tests instead then.